### DISCUSSION TITLE
0.35.0 beta

### DISCUSSION THREAD

**Discussion by intitni** [intitni/CopilotForXcode#592] [Announcements]
<img alt="instruction" src="https://raw.githubusercontent.com/intitni/intitni/a8b3dae8a3a3118fc22913dde331f229a0b762af/copilot-for-xcode-update-instruction.svg"  height="30">

- Adds a new chat tab to replace the old one. The old chat tab will not be converted to the new one automatically, so you will need to manually start a new chat.
  
  The new things are:
  - New UI 
    - The responses will make use of the full width of the window. The text contents will still have a max width so they are more readable when the window is too wide.
    - Reimplemnted the pin to bottom behavior.
  - New context picker that allows you to better control the context of the chat. 

    <img width="566" alt="new-context-picker" src="https://github.com/user-attachments/assets/c8da2be7-557b-496c-b5ea-7959a778bf25">

    You can add context by clicking the `+` button and remove context by clicking the `x` button. You can still use `@` to add a scope to the context, we will improve this part in the later betas.
  - Updated prompt format to better utilize prompt caching from OpenAI and Claude.
  - Apply button on the code block to apply the suggestions to the target file or the current file. The chat will also prompt the chat models to indicates the target file of a code block. (The implementation is still in progress, so it may not work as expected and may take some time to complete.)

    <img width="566" alt="apply-button" src="https://github.com/user-attachments/assets/1146e850-90c4-4e03-ab0f-cf01a6f273aa">

    In this example, clicking the apply button will generate the new file content with the suggestion for the file `Contents.swift`. The new content will be presented in the modification window.

  - You can send multiple message at the same time withot waiting for the response.
  - The custom command will behave differently now. The `extra system prompt` will now be injected into the context as a `topic`. `Topic`s will be put into the system prompt.

<hr /><em>This discussion was created from the release <a href='https://github.com/intitni/CopilotForXcode/releases/tag/0.35.0.beta'>0.35.0 beta</a>.</em>

**Comment by ajason** [comment]
The new chat is unresponsive, while the legacy chat functions normally. 
Chat model: GitHub Copilot (POC).

**Reply by intitni** [reply]
Oh no, I forgot to give it a try.. Thanks for reporting.

**Reply by intitni** [reply]
Fixed in beta 2
